src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/GlobalSingleInstanceActivationMaintainer.cs

```diff
@@ -250,40 +250,35 @@ private Task RunBatchedDemotion(List<Tuple<GrainId, KeyValuePair<ActivationId, I
                     responses[j] = batchResponses[j][i];
 
                 // response processor
-                var tracker = new GlobalSingleInstanceResponseTracker(responses, address.Grain, logger);
-
-                // since the tracker has all responses already, it is in completed state
-                if (!tracker.Task.IsCompleted)
-                    ProtocolError(address, "assertion error: tracker did not complete");
-
-                var outcome = tracker.Task.Result;
+                var outcomeDetails = GlobalSingleInstanceResponseTracker.GetOutcome(responses, address.Grain, logger);
+                var outcome = outcomeDetails.State;
 
                 if (logger.IsVerbose2)
                     logger.Verbose2("GSIP:M {0} Result={1}", address.Grain.ToString(), outcome.ToString());
```

Comment:
if we add the `ToString()` that prints a remotely owned activation address we can also use `outcomeDetails.ToString()` here.